src/components/Form/Form.tsx

```diff
@@ -34,6 +35,8 @@ const Form = <T extends FormDetails>({
   const [isLoading, setIsLoading] = useState(!!asyncGetDefaults);
   const [state, dispatch] = useAutoSaveReducer<T>(formReducer, initial);
 
+  const [isDirty, setIsDirty] = useState(false); // Added isDirty state
```

Comment:
remove obvious comments

```suggestion
  const [isDirty, setIsDirty] = useState(false);
```